# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/0001-auth-mech_winbind_ntlm-use-mech_winbind_auth_initial.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 889f5450e42af998aa5eaf9bc96d97003433113e Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Wed, 12 Jan 2011 17:18:08 +0100
Subject: [PATCH] auth: Fixed mech_winbind_ntlm by using mech_winbind_auth_initial()

In changeset 9002 a new function named mech_winbind_auth_initial() was
introduced, but only the mech_winbind_spnego module part was changed to finally
use this function.
The mech_winbind_ntlm module part was not properly adapted which leads to a non
functioning NTLM authentication mechanism while using winbind, because it will
never call the `ntlm_auth` helper.

The simple fix is to replace mech_generic_auth_initial() with
mech_winbind_auth_initial() as auth_initial "hook" when setting up the
mech_winbind_ntlm module, like it is done in the mech_winbind_spnego module.

References:

- Changeset 9002: http://hg.dovecot.org/dovecot-2.0/diff/9d0037a997f4/src/auth/mech-winbind.c
---
 src/auth/mech-winbind.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/auth/mech-winbind.c b/src/auth/mech-winbind.c
index 6009918..4299191 100644
--- a/src/auth/mech-winbind.c
+++ b/src/auth/mech-winbind.c
@@ -335,7 +335,7 @@ const struct mech_module mech_winbind_ntlm = {
 	.passdb_need = MECH_PASSDB_NEED_NOTHING,
 
 	mech_winbind_ntlm_auth_new,
-	mech_generic_auth_initial,
+	mech_winbind_auth_initial,
 	mech_winbind_auth_continue,
 	mech_generic_auth_free
 };
-- 
1.6.6.2

